### ISSUE TITLE
Repeated nested messages doesn't work in GET requests queries.

### ISSUE THREAD

**Issue by DmitriyMV** [grpc-ecosystem/grpc-gateway#492]
Given the 
```
service MyService {
    rpc myMethod (myMethodRequest) returns (myMethodResponse) {
        option (google.api.http) = {
            get: "/method/all"
        };
    }
}

message myMethodRequest {
    repeated Scoped scope = 1;

    message Scoped {
        oneof condition {
            ById by_id = 1;
        }
        message ById {
            uint32 id = 1;
        }
    }
}

message myMethodResponse {
    repeated uint32 ids = 1;
}
```

with GET request
```
localhost:8080/method/all?scope.by_id.id=3
```

returns an error

```
unexpected repeated field in scope.by_id.id
```

**Comment by rogerhub** [comment]
As far as I can tell, there isn't a way to specify a repeated message field using GET query parameters alone. Only repeated primitive fields (strings, ints, bools, floats, etc) are supported in query parameters. See [runtime/query.go](https://github.com/grpc-ecosystem/grpc-gateway/blob/master/runtime/query.go).

Can you use POST instead?

**Comment by achew22** [comment]
I think @rogerhub's answer is probably the best way to go here. I'm not sure what the standard syntax for repeated fields in a query string would be. I guess the PHP format? I'm not sure though and think you would probably be happier with this as a different verb.

**Comment by sum2000** [comment]
I agree with the proposal and willing to take solve this, I am new to this repo so what will be a good starting point to solve this?

**Comment by johanbrandhorst** [comment]
Hi @sum2000, thanks for showing an interest in contributing! I think all the query parameter parsing happens in https://github.com/grpc-ecosystem/grpc-gateway/blob/5ebe3a8e01b56f67d5f0c6da3a274639e5e1a376/runtime/query.go#L47, so that would be the place to start looking. The first thing to do would be to add a new test case to https://github.com/grpc-ecosystem/grpc-gateway/blob/master/runtime/query_test.go#L85 and see if you can make it pass. Let me know if you need anymore pointers!

**Comment by sum2000** [comment]
> Hi @sum2000, thanks for showing an interest in contributing! I think all the query parameter parsing happens in
> 
> https://github.com/grpc-ecosystem/grpc-gateway/blob/5ebe3a8e01b56f67d5f0c6da3a274639e5e1a376/runtime/query.go#L47
> , so that would be the place to start looking. The first thing to do would be to add a new test case to [/runtime/query_test.go@`master`#L85](https://github.com/grpc-ecosystem/grpc-gateway/blob/master/runtime/query_test.go#L85) [](/grpc-ecosystem/grpc-gateway/blob/HEAD@{2019-09-10T16:26:39Z}/runtime/query_test.go#L85) and see if you can make it pass. Let me know if you need anymore pointers!

Thanks, will do! Is there a discord or chat for grpc-gateway ?

**Comment by johanbrandhorst** [comment]
We have a channel on the gophers slack: #grpc-gateway after joining https://invite.slack.golangbridge.org/.

**Comment by npuichigo** [comment]
any updates? quite a good feature to support

**Comment by johanbrandhorst** [comment]
I don't think there's been any work on this recently unfortunately, but if you'd like to see this feature in the gateway, please feel free to have a go at implementing it. My advice above is still relevant.

**Comment by mauza** [comment]
I just ran into this same issue. We created an ordering struct `Page` so you could specify multiple (repeated) fields to order the response by on the GetPersons endpoint. The easiest way to fix this is to make it a Post to "GetPersons" but that just seems silly.
